tests/ci.rs

```diff
@@ -347,15 +368,104 @@ fn core() {
+                        if churns.iter().all(|b| *b) {
+                            break;
+                        }
                     }
+
+                    _ => (),
                 }
+            } else {
+                panic!("Timeout");
+            }
+        }
+    }
 
-                _ => (),
+    {
+        // message from quorum - 1 group members
+        let client = TestClient::new(nodes.len(), event_sender.clone());
+        let data = gen_plain_data();
+
+        loop {
+            if let Some(test_event) = recv_with_timeout(&event_receiver, Duration::from_secs(5)) {
+                match test_event {
+                    TestEvent(index, Event::Connected) if index == client.index => {
+                        unwrap_result!(client.client
+                                             .send_put_request(Authority::ClientManager(*client.name()), data.clone()));
+                    }
+                    TestEvent(index,
+                              Event::Request(RequestMessage{ src: Authority::Client{ .. },
+                                                                    dst: Authority::ClientManager(name),
+                                                                    content: RequestContent::Put(data, id) })) => {
+                        unwrap_result!(nodes[index].node.send_put_request(Authority::ClientManager(name),
+                                                                          Authority::NaeManager(data.name().clone()),
+                                                                          data.clone(),
+                                                                          id.clone()));
+                    }
+                    TestEvent(index, Event::Request(ref msg)) => {
+                        if let RequestContent::Put(_, ref id) = msg.content {
+                            if index < QUORUM_SIZE - 1 {
+                                unwrap_result!(nodes[index].node.send_put_failure(msg.dst.clone(),
+                                                                                  msg.src.clone(),
+                                                                                  msg.clone(),
+                                                                                  vec![],
+                                                                                  id.clone()));
+                            }
+                        }
+                    }
+                    TestEvent(_index,
+                              Event::Response(ResponseMessage{ content: ResponseContent::PutFailure{ .. },
+                                                                      .. })) => {
+                        panic!("Unexpected response.");
+                    }
+                    _ => (),
+                }
+            } else {
+                break;
+            }
+        }
+    }
+
+    {
+        // message from quorum - 1 group members
```

Comment:
But exactly `QUORUM_SIZE` responses should already suffice to cause the event, right? It might make sense to test this case, i. e. send 5 responses instead of 8 and assert that the event still occurs.